homeassistant/components/huum/__init__.py

```diff
@@ -0,0 +1,46 @@
+"""The Huum integration."""
+from __future__ import annotations
+
+import logging
+
+import aiohttp
+from huum.huum import Huum
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Huum from a config entry."""
+    username = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+
+    huum = Huum(username, password, session=async_get_clientsession(hass))
+
+    try:
+        await huum.status()
+    except aiohttp.ClientError as err:
+        _LOGGER.error("Could not log in to Huum with given credentials")
+        raise ConfigEntryNotReady(
+            "Could not log in to Huum with given credentials"
+        ) from err
```

Comment:
`ConfigEntryAuthFailed` is usually raised for authentication issues as, IIRC, it offers the user a way to start the reauth flow. `ConfigEntryNotReady` will just keep retrying forever while showing the error message on the integration page without actionable items.

Reply:
This is a valid point, but I like to have reauth added in a later PR as it's not needed for the initial integration PR. But I did not mention this yet, thanks for reminding.

Reply:
Is it enough to just raise `ConfigEntryAuthFailed` or does there need to be more logic in place for things to work? I mean, I can just check the status code of the response and if it is 401 and then throw that error.

Reply:
That should go in a follow up PR, it should also include a config flow where the user can reenter their password and restore their entry again

Reply:
I guess that would need to be checked every place where an API call is made, so when checking status, turning on/off etc as well right?

Reply:
I mean you have that concentrated in 2 places so its not a big of a deal :)

Reply:
But that is enough then, no other code is needed, just raise the exception, and done?

I guess I should start using custom exceptions coming from my library instead of "leaking" `aiohttp` exceptions as well. I will try to make those changes as well.

Reply:
Good point on keeping it simple for the initial PR 👍 

For exception handling, it's indeed better to raise library specific custom exceptions, but you can still use "raise .. from ex" to keep the original exception around.

Reply:
I'll keep it simple for now and skip implementing re-auth logic, I will however implement the custom exceptions so that it is easier to implement the re-auth when the time comes

Reply:
With that I see this matter resolved for now 😄